doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,495 @@
+
+
+_**Note 1:** bridge is not removed during the PF configuration reset_
+
+_**Note 2:** information about bridges is not reported in the status field of the `SriovNetworkNodeState` CR_
+
+#### Externally managed NIC flow
+
+In this case, **sriov-network-operator** handles announcement of SRIOV resources with device plugin and preparation of `NetworkAttachmentDefinition` CR.
+
+1. User configures PFs, creates VFs, creates bridge and attaches PFs to the bridge with custom scripts.
+
+2. User creates `SriovNetworkNodePolicy` where:
+    * eswitch mode set to `switchdev`
+    * numVFs set to match amount of precreated VFs
+    * `externallyManaged` set to true for PFs
+
+3. `sriov-network-config-daemon` configure VFs
+
+3. SRIOV resources are announced by the Device plugin
+
+4. User creates `OVSNetwork` or `BridgeNetwork` CR to create `NetworkAttachmentDefinition` CR, which relies on resources announced by the Device Plugin
+
+### API Extensions
+
+#### Environment variables for the operator
+
+| Variable                       | Description                                                       |
+| ---                            | ---                                                               |
+| `OVS_CNI_IMAGE`                | contains full image name for `ovs-cni`                            |
+| `ACCELERATED_BRIDGE_CNI_IMAGE` | contains full image name for `accelerated-bridge-cni`             |
+| `OVSDB_SOCKET_PATH`            | path to the OVSDB socket, used to path to the sriov config daemon |
+
+#### Daemon command line args
+
+`--ovsdb-socket-path` - path to the OVSDB socket, defaults to `/var/run/openvswitch/db.sock`
+
+#### OVSNetwork CRD `new`
+
+```golang
+// OvsNetworkSpec defines the desired state of OvsNetwork
+type OvsNetworkSpec struct {
+	// Namespace of the NetworkAttachmentDefinition custom resource
+	NetworkNamespace string `json:"networkNamespace,omitempty"`
+	// OVS Network device plugin endpoint resource name
+	ResourceName string `json:"resourceName"`
+	// Capabilities to be configured for this network.
+	// Capabilities supported: (mac|ips), e.g. '{"mac": true}'
+	Capabilities string `json:"capabilities,omitempty"`
+	// IPAM configuration to be used for this network.
+	IPAM string `json:"ipam,omitempty"`
+	// name of the OVS bridge, if not set OVS will automatically select bridge
+	// based on VF PCI address
+	Bridge string `json:"bridge,omitempty"`
+	// VlanTag to assign for the OVS port
+	VlanTag uint `json:"vlan,omitempty"`
+	// Mtu for the OVS port
+	MTU int `json:"mtu",omitempty`
+	// Trunk configuration for the OVS port
+	Trunk []*OVSTrunkConfig `json:"trunk,omitempty"`
+	// OpenFlow port number in range 1 to 65,279
+	OfportRequest uint `json:"ofport_request,omitempty"`
+	// The type of interface on ovs.
+	InterfaceType string `json:"interface_type,omitempty"`
+	// Path to the configuration file on the host
+	ConfigurationPath string `json:"configuration_path,omitempty"`
```

Comment:
can you give an example for this?
not sure I want to expose something like that to the user